sql/parser/eval.go

```diff
@@ -136,13 +139,13 @@ var binOps = map[binArgs]binOp{
 	binArgs{Plus, dateType, intervalType}: {
 		returnType: DummyTimestamp,
 		fn: func(left Datum, right Datum) (Datum, error) {
-			return DTimestamp{Time: left.(DDate).Add(right.(DInterval).Duration)}, nil
+			return DTimestamp{Time: left.(DDate).toTime().Add(right.(DInterval).Duration)}, nil
```

Comment:
I think we need to remove this type of casting of date to time in the
binary operations. I'm going to remove them.

On Mon, Oct 12, 2015 at 11:36 AM Ben Darnell <notifications@github.com>
wrote:

> In sql/parser/eval.go
> <https://github.com/cockroachdb/cockroach/pull/2811#discussion_r41769150>:
>
> > @@ -136,13 +139,13 @@ var binOps = map[binArgs]binOp{
> >  	binArgs{Plus, dateType, intervalType}: {
> >  		returnType: DummyTimestamp,
> >  		fn: func(left Datum, right Datum) (Datum, error) {
> > -			return DTimestamp{Time: left.(DDate).Add(right.(DInterval).Duration)}, nil
> > +			return DTimestamp{Time: left.(DDate).toTime().Add(right.(DInterval).Duration)}, nil
>
> toTime always uses UTC; a date plus or minus an interval or timestamp
> should be computed in the session time zone. (and we need some tests that
> expose this)
>
> toTime is rarely useful and it looks like we only ever use it correctly
> when we want to pull the (year, month, day) fields out of a value so we
> should probably replace toTime() with a toDate() (int, int, int) method.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/2811/files#r41769150>.
>